server/scores.js

```diff
@@ -4,27 +4,80 @@ const express = require('express')
 const Promise = require('bluebird')
 const { MongoClient, ObjectID } = require('mongodb')
 const { authroizationMiddlware } = require('@first-lego-league/ms-auth')
+const Configuration = require('@first-lego-league/ms-configuration')
 
 const DEFAULTS = require('./defaults')
 
 const mongoUrl = process.env.MONGO_URI || DEFAULTS.MONGO
 
 const router = express.Router()
 
+const STATUS = {
+  GOOD: 'ok',
+  CONFIGURATION_ERROR: 'config-error',
+  LOUD_FAIL: 'loud-fail'
+}
+
+const ERROR = {
+  TEAM_NUMBER: 'team number ',
+  SCORE: 'score ',
+  MATCH: 'match ',
+  NONE: ''
+}
+
 const connectionPromise = MongoClient
   .connect(mongoUrl, { promiseLibrary: Promise, useNewUrlParser: true })
   .then(client => client.db().collection('scores'))
 
+function _validateScore (score) {
+  let retError = { 'status': STATUS.GOOD, 'errors': ERROR.NONE }
+
+  Configuration.get('autoPublish').then(autoPublishSetting => {
+    const publishBool = autoPublishSetting
+    return publishBool
+  }).then(result => {
+    score.published = result
+  }).catch(() => {
+    retError.status = STATUS.CONFIGURATION_ERROR
+    return retError
+  })
+
+  if (typeof score.teamNumber !== 'number') { retError.errors += ERROR.TEAM_NUMBER }

```

Comment:
This is not a good way to work with promises. You shouldn't run synchronous calls at a promise, because they might not be called i in there correct order. Put everything in the promise, or the `then` functions and return it. Don't catch the error here but rather let it go over to the router and deal with it there. And assume no configuration errors.